python/pyspark/sql/session.py

```diff
@@ -556,8 +556,10 @@ def _create_from_pandas_with_arrow(self, pdf, schema, timezone):
         pdf_slices = (pdf[start:start + step] for start in xrange(0, len(pdf), step))
 
         # Create Arrow record batches
+        runner_conf = {"spark.sql.execution.pandas.arrowSafeTypeConversion":
+                       self._wrapped._conf.arrowSafeTypeConversion()}
         batches = [_create_batch([(c, t) for (_, c), t in zip(pdf_slice.iteritems(), arrow_types)],
-                                 timezone)
+                                 timezone, runner_conf)
```

Comment:
here, it doesn't really make sense to use a `runner_conf`, so it would be better to just pass in the flag